### ISSUE TITLE
Feature Request: allow currently mandatory, but syntactically redundant, braces to be optional

### ISSUE THREAD

**Issue by DavidArno** [dotnet/csharplang#181]
- [ ] Proposal added (draft created at [Redundant-brace-removal.md](https://github.com/DavidArno/csharplang/blob/master/proposals/Redundant-brace-removal.md), pull request [#180](https://github.com/dotnet/csharplang/pull/180))
- [ ] Championed
- [ ] Discussed in LDM
- [ ] Decision in LDM
- [ ] Finalized (done, rejected, inactive)
- [ ] Spec'ed

This proposal pulls together a couple of related proposals around removing the need to use `{}`, rather than `;` to terminate a declaration, along with two new opportunities to do so, in the following areas:
* [Single namespace in a file declarations](https://github.com/dotnet/csharplang/issues/137#issuecomment-281794438)
* [Constructors that just call `this` or `base`](https://github.com/dotnet/roslyn/issues/17306)
* Empty interface declarations
* Read-only auto-properties

**Comment by HaloFour** [comment]
Not a fan of the proposed auto-implemented property syntax:

```cs
public string Name get;
```

**Comment by DavidArno** [comment]
@HaloFour,

That's a shame, as that is my favourite of the four. The `{ get; }` syntax bugs me every time I have to write/read it. Different strokes...

**Comment by alrz** [comment]
Nit: you've conflated using statements and using directives. similarly, namespace declarations vs. namespace directives.

**Comment by JamesFaix** [comment]
Overall I like this idea, but also disagree with the auto-property component.  It can help disambiguate method or constructor bodies with no implementation by intention, vs bodies where the creator forgot to implement a stub or `throw NotImplementedException`.

**Comment by DavidArno** [comment]
@JamesFaix & @HaloFour,

I'd value your opinions here. As I seem in a minority of one on the auto-property change, should I update the proposal and submit a new PR without it, or leave it as a discussion point?

**Comment by eyalsk** [comment]
@DavidArno Is it really redundant? 

To me `public string Name get;` reads almost as if I declared a variable and it might be confusing, I mean, a single space dictates whether this is a variable or not.

There's so many ways to define properties that it's a bit overwhelming and there's [more ](https://github.com/dotnet/csharplang/issues/133) on the way.. 

Anyway, what's wrong with braces? I'm probably one of few that really like them: 

```
if (text == null)
{
	throw new ArgumentNullException(nameof(text));
}
```

😄

**Comment by DavidArno** [comment]
Closing as this proposal is mainly drawing negative feedback.

**Comment by eyalsk** [comment]
@DavidArno I don't know whether you directed this reply at me but it's pretty rude to say that it's negative just because we share different options.

You didn't even reply to me... if anything I'd say that's a negative and unacceptable attitude.

**Comment by jnm2** [comment]
@eyalsk I thought your questions were rhetorical. I don't think he realized that would communicate rudeness to anyone- and I didn't think it was rude. It's kind of awkward to handle negative feedback; it can kind of feel like everyone ganging up on you.

**Comment by eyalsk** [comment]
@jnm2 

> I thought your questions were rhetorical.

It was, when I said he didn't reply to me I meant that he decided to close the post right after I reply and didn't bother to say something positive before that and all I got is that my reply was negative, period.

>  I don't think he realized that would communicate rudeness to anyone- and I didn't think it was rude. 

Well, that's your opinion, I think it was **extremely** rude, to the point it disgust me! we probably interpret things differently.

>  It's kind of awkward to handle negative feedback; it can kind of feel like everyone ganging up on you.

I really don't understand this, why would you interpret disagreement as negative feedback? what's awkward is to close posts that aren't even a day long after few people said they disliked it.

**Comment by HaloFour** [comment]
@DavidArno

Only a couple of opinions.  Always worth eliciting more feedback.  While I don't prefer the property syntax you suggested, I'm fine with how this proposal works with interfaces and constructor bodies.  Possibly with empty classes, too.  Although then you may have to consider how the very similar syntax relates to records.

**Comment by jnm2** [comment]
@eyalsk I'm positive you're misinterpreting. "Negative feedback" means no more, no less, than a down vote. "Negative feedback" does not mean the feedback is bad, nor does it imply that the feedback was given in bad faith.

> we probably interpret things differently.

And that's okay!! :-) I just want to provide a data point that other interpretations exist. It's always better to choose the interpretation that gives the benefit of the doubt. Hanlon's razor etc.

**Comment by eyalsk** [comment]
@jnm2 Yes might be my bad interpretation but I think that it's more useful to say "Closing this because it seems like people dislike it." as opposed to "Closing as this proposal is mainly drawing negative feedback." which leaves a lot of place to imagination and/or doubt.

**Comment by jnm2** [comment]
@eyalsk

> Yes might be my bad interpretation but I think that it's more useful to say "Closing this because it seems like people dislike it." as opposed to "Closing as this proposal is mainly drawing negative feedback." which leaves a lot of place to imagination and/or doubt.

You may be right about that. We all learned something here... :-D

**Comment by eyalsk** [comment]
@jnm2 I hope. 😉

**Comment by DavidArno** [comment]
@eyalsk,

> I don't know whether you directed this reply at me but it's pretty rude to say that it's negative just because we share different opinions.

I'm really sorry: that wasn't my intention at all. The proposal itself has only down-votes, @HaloFour's comment that he wasn't a fan of the getter syntax was upvoted and the feedback from others was that they didn't like it either. To my mind that's negative feedback, which I've listened to as it was constructive as well. So I closed it, with the intention of removing the getter idea and creating a new PR and proposal.

I've been trying recently to be more polite and positive in my comments. I see I've a way to go if I'm still coming across as rude. So apologies once more for inadvertently being rude here.
